tsdb/engine/tsm1/cache.go

```diff
@@ -187,6 +189,15 @@ func (c *Cache) Snapshot() *Cache {
 		c.snapshotSize += uint64(Values(e.values).Size())
 	}
 
+	// Do deduplication a copy of the array.
```

Comment:
This loops is essentially the same thing that is happening on L183.  One piece that differs is the setting of `needSort`.